pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -166,22 +169,30 @@ func (b *AccessLogBuilder) setTCPAccessLog(push *model.PushContext, proxy *model
 
 func buildAccessLogFromTelemetry(push *model.PushContext, spec *model.LoggingConfig, forListener bool) []*accesslog.AccessLog {
 	als := make([]*accesslog.AccessLog, 0)
+	filter := buildAccessLogFilterFormTelemetry(spec)
 	for _, p := range spec.Providers {
 		switch prov := p.Provider.(type) {
 		case *meshconfig.MeshConfig_ExtensionProvider_EnvoyFileAccessLog:
 			al := buildEnvoyFileAccessLogHelper(prov.EnvoyFileAccessLog)
 			if forListener {
 				al.Filter = addAccessLogFilter()
```

Comment:
I haven't had a chance to think about this deeply, but this part is suspicious to me. The code is pretty subtle here, IIRC how it works is it logs any time the main access log would not be hit at all. This avoids requests with no logs at all.

In current form, I think this means if a log has filter `code == 200`, and the request fails to match a filter chain, we would log from the listener access log, which is not what we want?

So maybe we need to somehow merge their filter with the filter to avoid double logging... that sounds problematic though

Reply:
a listener access_log demo as following:
```yaml
access_log:
- filter:
    response_flag_filter:
      flags:
      - NR
  name: envoy.access_loggers.file
  typed_config:
    '@type': type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog
    log_format:
      text_format_source:
        inline_string: |
          [%START_TIME%] "%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%" %RESPONSE_CODE% %RESPONSE_FLAGS% %RESPONSE_CODE_DETAILS% %CONNECTION_TERMINATION_DETAILS% "%UPSTREAM_TRANSPORT_FAILURE_REASON%" %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)% "%REQ(X-FORWARDED-FOR)%" "%REQ(USER-AGENT)%" "%REQ(X-REQUEST-ID)%" "%REQ(:AUTHORITY)%" "%UPSTREAM_HOST%" %UPSTREAM_CLUSTER% %UPSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_REMOTE_ADDRESS% %REQUESTED_SERVER_NAME% %ROUTE_NAME%
    path: /dev/stdout
```
it will log when `NR`

Reply:
Example where this is wrong, IMO:

```
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: mesh-default
  namespace: istio-system
spec:
  accessLogging:
  - providers:
    - name: envoy
    filter:
      expression: "false"
```

This will still log. I would expect `false` to mean we *never* log

Reply:
Maybe what we want is:

```
AccessLogFilter_AndFilter {
  filters: [NR filter, user provided CEL filter]
}
```

Reply:
updated, PTAK